src/layers/convolution.rs

```diff
@@ -84,4 +104,44 @@ where
         output_dim[0] = self.out_channels;
         Tensor::from_shape_vec(output_dim, output_channels).unwrap()
     }
+
+    fn set_weights(&mut self, new_weights: Vec<f64>) {
+        let weight_count: usize = self
+            .kernels
+            .iter()
+            .map(|kernel| kernel.weights().len())
+            .sum();
+
+        if weight_count != new_weights.len() {
+            panic!("Loaded weight count does not match layer weight count.");
+        }
+
+        let mut new_iter = new_weights.iter();
+
+        self.kernels.iter_mut().for_each(|kernel| {
+            kernel
+                .weights_mut()
+                .iter_mut()
+                .for_each(|weight| *weight = val!(*new_iter.next().unwrap()))
+        });
```

Comment:
> Also changing the code I just suggested to (weight + 1.).into() will still pass all tests, is this expected?

Not sure what you mean here as I see no reason why it wouldn't pass all tests. Are you not just adding 1 to every weight?

Reply:
It passed all tests because there is no test for checking if state dict for a Convolution layer was completed correctly.
Either way, what I was trying to get at was that if I added 1 to the weights in this method then there would be a disparity between the layers' weights and the state dict that was passed. But again, there was no test coverage for this so it makes sense why it passed

Reply:
Gotcha. Will add a test for convolutional layers.